cmd/createcluster.go

```diff
@@ -386,43 +400,86 @@ func getValidators(dvs []tbls.TSS) ([]cluster.DistValidator, error) {
 	return vals, nil
 }
 
-// getOperators returns a list of operators from the provided secret keyshares.
-func getOperators(numNodes int, clusterDir string, insecureKeys bool, shareSets [][]*bls_sig.SecretKeyShare) ([]cluster.Operator, error) {
-	var peers []p2p.Peer
+// saveKeysToKeyManager writes validator keys to the provided keymanager addresses.
+func writeKeysToKeymanager(ctx context.Context, addrs []string, numNodes int, shareSets [][]*bls_sig.SecretKeyShare) error {
+	eg := new(errgroup.Group)
 	for i := 0; i < numNodes; i++ {
-		peer, err := newPeer(clusterDir, i)
-		if err != nil {
-			return nil, err
+		var secrets []*bls_sig.SecretKey
+		for _, shares := range shareSets {
+			secret, err := tblsconv.ShareToSecret(shares[i])
+			if err != nil {
+				return err
+			}
+			secrets = append(secrets, secret)
 		}
 
-		peers = append(peers, peer)
+		addr := addrs[i]
+		eg.Go(func() error {
+			err := keystore.PostKeysToKeymanager(ctx, addr, secrets)
```

Comment:
is this is slow call? why even do it concurrently? feels like a premature optimisation, rather just do sequentially.